sql/core/src/test/scala/org/apache/spark/sql/connector/DataSourceV2SQLSuite.scala

```diff
@@ -3627,6 +3629,54 @@ class DataSourceV2SQLSuiteV1Filter
     }
   }
 
+  def checkParquet(tableName: String, path: String): Unit = {
+    withTable(tableName) {
+      sql("CREATE TABLE " + tableName +
+        " (name STRING) USING PARQUET LOCATION '" + path + "'")
+      sql("INSERT INTO " + tableName + " VALUES('Bob')")
+      val df = sql("SELECT * FROM " + tableName)
+      assert(df.queryExecution.analyzed.exists {
+        case LogicalRelation(_: HadoopFsRelation, _, _, _) => true
+        case _ => false
+      })
+      checkAnswer(df, Row("Bob"))
+    }
+  }
+
+  test("SPARK-49211: V2 Catalog can support built-in data sources") {
+    // Reset CatalogManager to clear the materialized `spark_catalog` instance, so that we can
+    // configure a new implementation.
+    val table1 = FullQualifiedTableName(SESSION_CATALOG_NAME, "default", "t")
+    spark.sessionState.catalogManager.reset()
+    withSQLConf(
+      V2_SESSION_CATALOG_IMPLEMENTATION.key ->
+        classOf[V2CatalogSupportBuiltinDataSource].getName) {
+      withTempPath { path =>
+        checkParquet(table1.toString, path.getAbsolutePath)
+      }
+    }
+    val table2 = FullQualifiedTableName("testcat3", "default", "t")
+    withSQLConf(
+      "spark.sql.catalog.testcat3" -> classOf[V2CatalogSupportBuiltinDataSource].getName) {
+      withTempPath { path =>
+        checkParquet(table2.toString, path.getAbsolutePath)
+      }
+    }
+  }
+
+  test("SPARK-49211: V2 Catalog support CTAS") {
+    val table2 = FullQualifiedTableName("testcat3", "default", "t")
+    withSQLConf(
+      "spark.sql.catalog.testcat3" -> classOf[V2CatalogSupportBuiltinDataSource].getName) {
```

Comment:
can we also test custom `spark_catalog`?

Reply:
done